src/controllers/programs-controller.js

```diff
@@ -38,34 +40,74 @@ router.get("/byPrerequisites", async (req, res) => {
     try {
         const requestedPrerequisites = req.query.prerequisites;
 
-        if (!requestedPrerequisites) {
-            return res.status(400).json({ error: "Invalid or missing prerequisites in the request." });
-        }
-
-        // Based on the user's request of prerequisites (number of prerequisites),
-        // The system converts a single prerequisite into an array of size one.
         const prerequisitesArray = Array.isArray(requestedPrerequisites) ? requestedPrerequisites : [requestedPrerequisites];
 
-        // Checks if there are any invalid values included in the array.
-        const invalidPrerequisites = prerequisitesArray.filter(prerequisite => !["Math3B", "Math4", "Math5", "Physics1A", "Physics2", "Chemistry1", "Chemistry2", "Biology1", "Biology2"].includes(prerequisite));
+        // List of all valid prerequisites
+        const validPrerequisites = ["Math3B", "Math4", "Math5", "Physics1A", "Physics2", "Chemistry1", "Chemistry2", "Biology1", "Biology2", "Science2", "Civics1B", "History1B", "Language3", "SpecialRequirement"];
+        
+        const invalidPrerequisites = prerequisitesArray.filter(prerequisite => !validPrerequisites.includes(prerequisite));
+
+        // Map for higher category prerequisites
```

Comment:
Tests have been performed to see what happens when a user has already stated that he/she satisfies both Math3B and Math4